src/Job/StaticCacheFullBuildJob.php

```diff
@@ -0,0 +1,100 @@
+     */
+    public function getTitle()
+    {
+        return 'Generate a static pages for all URLs';
+    }
+
+    /**
+     * @return string
+     */
+    public function getSignature()
+    {
+        return md5(static::class);
+    }
+
+    public function setup()
+    {
+        parent::setup();
+        $this->URLsToProcess = $this->getAllLivePageURLs();
+        $this->URLsToCleanUp = [];
+        $this->totalSteps = ceil(count($this->URLsToProcess) / self::config()->get('chunk_size'));
+        $this->addMessage(sprintf('Building %s URLS', count($this->URLsToProcess)));
+        $this->addMessage(var_export(array_keys($this->URLsToProcess), true));
+    }
+
+    /**
+     * Do some processing yourself!
+     */
+    public function process()
+    {
+        $chunkSize = self::config()->get('chunk_size');
+        $count = 0;
+        foreach ($this->jobData->URLsToProcess as $url => $priority) {
+            if (++$count > $chunkSize) {
+                break;
+            }
+            $meta = Publisher::singleton()->publishURL($url, true);
+            if (!empty($meta['success'])) {
+                $this->jobData->ProcessedURLs[$url] = $url;
+                unset($this->jobData->URLsToProcess[$url]);
+            }
+        }
+        if (empty($this->jobData->URLsToProcess)) {
+            $trimSlashes = function ($value) {
+                return trim($value, '/');
+            };
+            $this->jobData->publishedURLs = array_map($trimSlashes, Publisher::singleton()->getPublishedURLs());
+            $this->jobData->ProcessedURLs = array_map($trimSlashes, $this->jobData->ProcessedURLs);
+            $this->jobData->URLsToCleanUp = array_diff($this->jobData->publishedURLs, $this->jobData->ProcessedURLs);
+            foreach ($this->jobData->URLsToCleanUp as $staleURL) {
+                $staleURL = $staleURL == '' ? '/' : $staleURL;
+                $purgeMeta = Publisher::singleton()->purgeURL($staleURL);
+                if (!empty($purgeMeta['success'])) {
+                    unset($this->jobData->URLsToCleanUp[$staleURL]);
+                }
+            }
+        };
+        $this->isComplete = empty($this->jobData->URLsToProcess) && empty($this->jobData->URLsToCleanUp);
+    }
+
+    /**
+     *
+     * @return array
+     */
+    protected function getAllLivePageURLs()
+    {
+        $urls = [];
+        $this->extend('beforeGetAllLivePageURLs', $urls);
+        $livePages = Versioned::get_by_stage(SiteTree::class, Versioned::LIVE);
+        foreach ($livePages as $page) {
+            if ($page->hasExtension(PublishableSiteTree::class) || $page instanceof StaticallyPublishable) {
+                $urls = array_merge($urls, $page->urlsToCache());
+            }
+        }
+
+        $this->extend('afterGetAllLivePageURLs', $urls);
+        // @TODO look here when integrating subsites
+        // if (class_exists(Subsite::class)) {
+        //     Subsite::disable_subsite_filter(true);
+        // }
```

Comment:
Do you want to do this now or later?

Reply:
Later - subsites is on our "nice to have list" and we need to get this merged and move on to other work

Reply:
Will be a connector module eventually :)